beacon_node/http_api/src/publish_attestations.rs

```diff
@@ -0,0 +1,292 @@
+    chain: &Arc<BeaconChain<T>>,
+    attestation: &Attestation<T::EthSpec>,
+    seen_timestamp: Duration,
+    network_tx: &UnboundedSender<NetworkMessage<T::EthSpec>>,
+) -> Result<(), Error> {
+    let attestation = chain
+        .verify_unaggregated_attestation_for_gossip(attestation, None)
+        .map_err(Error::Validation)?;
+
+    // Publish.
+    network_tx
+        .send(NetworkMessage::Publish {
+            messages: vec![PubsubMessage::Attestation(Box::new((
+                attestation.subnet_id(),
+                attestation.attestation().clone(),
+            )))],
+        })
+        .map_err(|_| Error::Publication)?;
+
+    // Notify the validator monitor.
+    chain
+        .validator_monitor
+        .read()
+        .register_api_unaggregated_attestation(
+            seen_timestamp,
+            attestation.indexed_attestation(),
+            &chain.slot_clock,
+        );
+
+    let fc_result = chain.apply_attestation_to_fork_choice(&attestation);
+    let naive_aggregation_result = chain.add_to_naive_aggregation_pool(&attestation);
+
+    if let Err(e) = fc_result {
+        Err(Error::ForkChoice(e))
+    } else if let Err(e) = naive_aggregation_result {
+        Err(Error::AggregationPool(e))
+    } else {
+        Ok(())
+    }
+}
+
+pub async fn publish_attestations<T: BeaconChainTypes>(
+    task_spawner: TaskSpawner<T::EthSpec>,
+    chain: Arc<BeaconChain<T>>,
+    attestations: Vec<Attestation<T::EthSpec>>,
+    network_tx: UnboundedSender<NetworkMessage<T::EthSpec>>,
+    reprocess_send: Option<Sender<ReprocessQueueMessage>>,
+    log: Logger,
+) -> Result<(), warp::Rejection> {
+    // Collect metadata about attestations which we'll use to report failures. We need to
+    // move the `attestations` vec into the blocking task, so this small overhead is unavoidable.
+    let attestation_metadata = attestations
+        .iter()
+        .map(|att| (att.data.slot, att.data.index))
+        .collect::<Vec<_>>();
+
+    // Gossip validate and publish attestations that can be immediately processed.
+    let seen_timestamp = timestamp_now();
+    let mut prelim_results = task_spawner
+        .blocking_task(Priority::P0, move || {
+            Ok(attestations
+                .into_iter()
+                .map(|attestation| {
+                    match verify_and_publish_attestation(
+                        &chain,
+                        &attestation,
+                        seen_timestamp,
+                        &network_tx,
+                    ) {
+                        Ok(()) => PublishAttestationResult::Success,
+                        Err(Error::Validation(AttestationError::UnknownHeadBlock {
+                            beacon_block_root,
+                        })) => {
+                            let Some(reprocess_tx) = &reprocess_send else {
+                                return PublishAttestationResult::Failure(Error::ReprocessDisabled);
```

Comment:
Just a note while reviewing  - was thinking whether we keep the original error, but realise it's probably not worth it
1. beacon processor are enabled by default
2. this is the only usage of `ReprocessDisabled`, if we get this we know we've got an `UnknownHeadBlock` - if we do re-procssing in more scenarios in the future it might be worth keeping the original error.